### PR TITLE
terminus-cli: init at 3.0.6

### PR DESCRIPTION
**Pull request by asimpson**
###### Motivation for this change
Adds [Terminus](https://pantheon.io/docs/terminus) to nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))

- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14066,4 +14066,10 @@
     github = "nigelgbanks";
     githubId = 487373;
   };
+  asimpson = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:14069]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -34757,4 +34757,6 @@ with pkgs;
   honeytail = callPackage ../servers/tracing/honeycomb/honeytail { };
 
   honeyvent = callPackage ../servers/tracing/honeycomb/honeyvent { };
+
+  terminus-cli = callPackage ../development/php-packages/terminus-cli { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:34761]: Please sort this into a fitting category and sort there alphabetically.

File: pkgs/development/php-packages/terminus-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    makeWrapper ${php}/bin/php $out/bin/terminus \
+      --add-flags "$out/libexec/terminus.phar" \
+      --prefix PATH : ${lib.makeBinPath [ unzip ]}
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A standalone utility for performing operations on the Pantheon Platform";
+    homepage = "https://pantheon.io/docs/terminus";
+    changelog = "https://github.com/pantheon-systems/terminus/blob/3.x/CHANGELOG.md";
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/terminus-cli/default.nix:29]:
```suggestion
    changelog = "https://github.com/pantheon-systems/terminus/blob/HEAD/CHANGELOG.md";
```

**Comment by asimpson** [comment]
Closing for now since I no longer need terminus but if someone else wants to open it back up, feel free.
